File: JavaScript/src/Employee.mjs
export default class Employee {
  /**
   * @readonly
   * @type {number}
   */
  rate

  /**
   * @private
   * @readonly
   * @type {boolean}
   */
  #separated

  /**
   * @private
   * @readonly
   * @type {boolean}
   */
  #retired

  /**
   * @param {number} rate
   * @param {boolean} separated
   * @param {boolean} retired
   */
  constructor(rate, separated, retired) {
    this.rate = rate
    this.#separated = separated
    this.#retired = retired
  }

  /**
   * @returns {number}
   */
  get rate() {
    return this.rate
  }

  /**
   * @returns {boolean}
   */
  isSeparated() {
    return this.#separated
  }

  /**
   * @returns {boolean}
   */
  isRetired() {
    return this.#retired
  }
}

File: cpp/src/Employee.cpp
#include "Employee.h"

Employee::Employee(int rate, bool separated, bool retired)
    : rate(rate), separated(separated), retired(retired) {}

int Employee::getRate() const {
    return rate;
}

bool Employee::isSeparated() const {
    return separated;
}

bool Employee::isRetired() const {
    return retired;
}